jobs/payment-jobs/tasks/electronic_funds_transfer_task.py

```diff
@@ -12,273 +12,136 @@
+        receipt_response = CFSService.apply_receipt(cfs_account, receipt_number, invoice_number)
+        receipt_json = receipt_response.json()
+        receipt = ReceiptModel(receipt_number=receipt_json.get('receipt_number', None),
+                               receipt_amount=amount,
+                               invoice_id=invoice.id,
+                               receipt_date=datetime.now(tz=timezone.utc))
+        receipt.flush()
 
+    @classmethod
+    def link_electronic_funds_transfers_cfs(cls):
+        """Replicate linked EFT's as receipts inside of CFS and mark invoices as paid."""
+        credit_invoice_links = cls._get_eft_credit_invoice_links_by_status(EFTCreditInvoiceStatus.PENDING.value)
+        for invoice, credit_invoice_link, cfs_account in credit_invoice_links:
+            try:
+                current_app.logger.debug(f'Create EFT Invoice Links in CFS - PayAccount: {invoice.payment_account_id}'
+                                         f'Id: {credit_invoice_link.id} - Amount: {credit_invoice_link.amount} '
+                                         f'- Invoice Id: {invoice.id}')
+                invoice_reference = InvoiceReferenceModel.find_by_invoice_id_and_status(
+                    credit_invoice_link.invoice_id, InvoiceReferenceStatus.ACTIVE.value
+                )
+                invoice_reference.status_code = InvoiceReferenceStatus.COMPLETED.value
+                invoice_reference.flush()
+                # Note: Not creating the entire EFT as a receipt because it can be mapped to multiple CFS accounts.
+                # eft_credit_invoice_links table should reflect exactly what's in CAS.
+                receipt_number = f'EFT-CIL-{credit_invoice_link.id}'
                 CFSService.create_cfs_receipt(
                     cfs_account=cfs_account,
                     rcpt_number=receipt_number,
-                    rcpt_date=payment.payment_date.strftime('%Y-%m-%d'),
-                    amount=payment.invoice_amount,
-                    payment_method=payment_account.payment_method,
+                    rcpt_date=credit_invoice_link.created_on.strftime('%Y-%m-%d'),
+                    amount=credit_invoice_link.amount,
+                    payment_method=PaymentMethod.EFT.value,
                     access_token=CFSService.get_token(PaymentSystem.FAS).json().get('access_token'))
-
-                # apply receipt to cfs_account
-                total_invoice_amount = cls._apply_electronic_funds_transfers_to_pending_invoices(
-                    eft_short_name, receipt_number)
-                current_app.logger.debug(f'Total Invoice Amount : {total_invoice_amount}')
-
-                eft_credit.remaining_amount = eft_credit.amount - total_invoice_amount
-
-                eft_short_name.save()
-
+                cls._apply_eft_receipt_to_invoice(
+                    cfs_account, invoice, invoice_reference.invoice_number, receipt_number, credit_invoice_link.amount)
```

Comment:
Is it safe to assume `invoice.total = credit_invoice_link.amount`? or is that a bad assumption